internal/server/polling/terraform.go

```diff
@@ -38,6 +61,114 @@ func (s *Server) getSource(ctx context.Context, tf *v1alpha2.Terraform) (*source
 	return obj, nil
 }
 
+func (s *Server) createTerraform(ctx context.Context, originalTF *infrav1.Terraform, originalSource *sourcev1b2.GitRepository, branch string, prID string) error {
+	tf := originalTF.DeepCopy()
+
+	tf.ResourceVersion = ""
+
+	tfName := s.createObjectName(tf.ObjectMeta.Name, branch, prID)
+
+	tf.SetName(tfName)
+
+	tf.SetLabels(s.createLabels(tf.Labels, branch, prID))
+
+	msg := fmt.Sprintf("Terraform object %s in the namespace %s", tfName, tf.ObjectMeta.Namespace)
+
+	source, err := s.createSource(ctx, originalSource, branch, prID)
+	if err != nil {
+		return fmt.Errorf("unable to create Source for %s: %w", msg, err)
+	}
+
+	tf.Spec.SourceRef.Name = source.ObjectMeta.Name
```

Comment:
This relies on the (safe) assumption that the copy of the source must be in the same namespace as the original source. A comment explaining why this assumption is safe would help people read the code -- or you could just defensively assign the namespace, so people don't even have to pause and think about it.

Reply:
What could be a reason not to put branch-based sources to the same namespace as the original source?